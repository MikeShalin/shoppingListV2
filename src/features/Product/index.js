/** @flow **/
import React from 'react'
import {
  List,
  Image,
  Button,
} from 'semantic-ui-react'

import { getIcons } from 'Utils/helpers'

type TProps = {
  name: string,
  children: any,
  date: string,
}

const ListExampleDivided = ({
                              name,
                              children,
                              date,
                            }: TProps) => (
  <List.Item>
    <Image avatar src={getIcons(name)}/>
    <List.Content>
      <List.Header as='a'>{children}</List.Header>
      <List.Description as='a'>Updated {date} ago</List.Description>
    </List.Content>
    <List.Content floated='right'>
      <Button negative>âœ–</Button>
    </List.Content>
  </List.Item>
)

export default ListExampleDivided
