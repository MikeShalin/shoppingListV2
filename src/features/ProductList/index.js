/** @flow **/
import React from 'react'
import { List } from 'semantic-ui-react'
import {
  compose,
  lifecycle,
  type HOC,
} from 'recompose'
import { inject, observer } from 'mobx-react'

import Product from 'Features/Product'

const ListExampleDivided = () => (
  <List divided relaxed>
    <Product name='rachel' date='10'>
      Semantic-Org/Semantic-UI
    </Product>
    <Product name='lol' date='22'>
      Semantic-Org/Semantic-UI-Docs
    </Product>
    <Product name='meow' date='34'>
      Semantic-Org/Semantic-UI-Meteor
    </Product>
  </List>
)

const composed: HOC<*, {}> = compose(
  inject('productList'),
  observer,
  lifecycle({
    componentDidMount(){
      console.log(JSON.parse(this.props.productList.getList()))
    }
  })
)

export default composed(ListExampleDivided)
