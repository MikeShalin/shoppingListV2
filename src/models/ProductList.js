import { types } from 'mobx-state-tree'

const ExecutionDataItem = types.model({
  id: types.number,
  start_time: types.string,
  finish_time: types.string,
  program_version: types.number,
})

const ProductList = types
  .model('Store', {
    list: types.optional(types.array(ExecutionDataItem), []),
    error: types.union(types.string, types.null),
  })
  .actions(self => ({
    handlerSubmit() {
      console.log('it submit ', self.productName)
      // self.productName = location
    },
    handleChange(target) {
      self.productName = target
    },
  }))

export default ProductList.create({
  list: [],
  error: null,
})
