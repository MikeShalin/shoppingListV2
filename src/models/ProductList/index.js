import { types } from 'mobx-state-tree'
import _ from 'lodash'

import { getItem, setItem } from 'Utils/localStorage'
import { key } from 'Config'

const Product = {
  id: types.number,
  name: types.string,
  icon: types.string,
  date: types.string,
}

const ProductList = types
  .model('Store', {
    list: types.frozen(Product),
    error: types.union(types.string, types.null),
  })
  .actions(self => ({
    removeProductOfList(id) {
      const list = JSON.parse(getItem(key))
      self.list = _.omit(list, id)
      setItem({ key, value: JSON.stringify(self.list) })
      return true
    },
    addToProductList(product) {
      const value = {...JSON.parse(getItem(key)), ...product}
      setItem({ value: JSON.stringify(value), key })
      self.list = value
      return true
    },
    handleChange(target) {
      self.productName = target
    },
  }))

export default ProductList
