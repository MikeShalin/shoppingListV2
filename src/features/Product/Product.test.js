import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Product from './index'
import ProductList from 'Models/ProductList'
import { product } from 'Fixtures'

const productList = ProductList.create(product)

describe('ProductList component', () => {
  it('matches early snapshots', () => {
    const props = {
      ...product.list,
      getUpdatedTime: () => 'a few seconds ago',
    }
    const tree = shallow(
      <Product.wrappedComponent productList={productList} {...props} />
    )
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})