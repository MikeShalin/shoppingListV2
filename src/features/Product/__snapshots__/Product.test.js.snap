// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList component matches early snapshots 1`] = `
<ListExampleDivided
  date="Sat Jan 05 2019 10:41:54 GMT+0400 (Самарское стандартное время)"
  getUpdatedTime={[Function]}
  icon="product"
  id={1}
  name="product"
  productList={
    Object {
      "error": null,
      "list": Object {
        "date": "Sat Jan 05 2019 10:41:54 GMT+0400 (Самарское стандартное время)",
        "icon": "product",
        "id": 1,
        "name": "product",
      },
    }
  }
/>
`;
